core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/common/AkkaActorService.scala

```diff
@@ -17,10 +17,11 @@ class AkkaActorService(val id: ActorVirtualIdentity, actorContext: ActorContext)
 
   def parent: ActorRef = actorContext.parent
 
-  var getAvailableNodeAddressesFunc: () => Array[Address] = () => Array.empty
+  var addresses: Array[Address] = Array()
```

Comment:
If the addresses are only used by the controller, can we remove them from AkkaActorService and use ControllerConfig?

Reply:
this information is used in `PhysicalOp.build`, where `ControllerConfig` is not available. I suggest we keep it in the service as it is also meaningful for a runtime service to manage it, instead of referring back to config (which is constant).